liberty-managed/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -1201,6 +1218,343 @@ private void throwDeploymentException(String applicationName) throws DeploymentE
+                           // We could even embed the log line via a fake cause:
+                           // throwable.initCause(new RuntimeException(message))
+                           xCon = clazz.getConstructor();
+                           thisServerThrowable = (Throwable) xCon.newInstance();
+                           thisServerThrowable.initCause(new RuntimeException(x.logMsg));
+                           
+                       } catch (NoSuchMethodException e3) {
+                           log.warning("Arquillian container could not load external Throwable 1, from: " + x);
+                           thisServerThrowable = new UnloadableLogError(x.logMsg);
+                           
+                       }
+                   }
+               }
+           } else { // Cause is not null
+               try {
+                   // Ideally we want to add the FFDC log line
+                   xCon = clazz.getConstructor(String.class, Throwable.class);
+                   thisServerThrowable = (Throwable) xCon.newInstance(new Object[] { x.logMsg, cause });
+                   
+               } catch (NoSuchMethodException e) {
+                   try {
+                       // Some 'old' errors can add the cause afterwards
+                       xCon = clazz.getConstructor(String.class);
+                       thisServerThrowable = (Throwable) xCon.newInstance(new Object[] { x.logMsg });
+                       thisServerThrowable.initCause(cause);
+                       
+                   } catch (NoSuchMethodException e2) {
+                       try {
+                           // We are prepared to loose the log line to get the correct exception
+                           xCon = clazz.getConstructor(Throwable.class);
+                           thisServerThrowable = (Throwable) xCon.newInstance(new Object[] { cause });
+                           
+                       } catch (NoSuchMethodException e3) {
+                           // We could not get the right result;
+                           log.warning("Arquillian container could not load external Throwable 2, from: " + x);
+                           thisServerThrowable = new UnloadableLogError(x.logMsg, cause);
+                       }
+                   }
+               }
+
+           }
+
+       } catch (InstantiationException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 3, from: " + x);
+       } catch (IllegalAccessException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 4, from: " + x);
+       } catch (IllegalArgumentException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 5, from: " + x);
+       } catch (InvocationTargetException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 6, from: " + x);
+       }
```

Comment:
Might want to change `e.getMessage()` to `e.toString()` so that it includes the exception class name in the log message.

Reply:
Done